localstack/services/cloudformation/models/cloudformation.py

```diff
@@ -76,3 +77,38 @@ def get_nested_stack_params(params, **kwargs):
                 "parameters": get_nested_stack_params,
             }
         }
+
+
+class CloudFormationMacro(GenericBaseModel):
+    @staticmethod
+    def cloudformation_type():
+        return "AWS::CloudFormation::Macro"
+
+    def get_physical_resource_id(self, attribute=None, **kwargs):
+        return self.props.get("Name")
+
+    def fetch_state(self, stack_name, resources):
+        return get_cloudformation_store().macros.get(self.props.get("Name"))
+
+    @classmethod
+    def get_deploy_templates(cls):
+        def _store_macro(resource_id, resources, resource_type, func, stack_name):
+            resource = resources[resource_id]
+            properties = resource["Properties"]
+            name = properties["Name"]
+            get_cloudformation_store().macros[name] = properties
```

Comment:
Hm interesting. At first I was actually suggesting to remove this, but it seems CloudFormation is doing some weird internal saving here as well, since that's the only way to "register" a macro to the account. Kinda wondering why they're not just using the Extensions for this as well :thinking: